tests/container_tests/test_zip_container.py

```diff
@@ -559,3 +559,260 @@ def func():
 
             self.assertEqual(p1.exitcode, 0)
             self.assertEqual(p2.exitcode, 0)
+
+
+class TestZipHandlerListNoDirectory(unittest.TestCase):
+    def setUp(self):
+        # The following zip layout is created for all the tests
+        # The difference is despite showing in the following layout for
+        # readabilty, the directories are not included in the zip
+        # outside.zip
+        # | - testdir1
+        # | - | - testfile1
+        # | - | - testdir2
+        # | - | - | - testfile2
+        # | - testdir3
+        # |   | - testfile3
+        # | - testfile4
+
+        self.test_string = "this is a test string\n"
+        self.fs_handler = chainerio.create_handler("posix")
+
+        # the most outside zip
+        self.zip_file_name = "outside.zip"
+
+        # nested zip and nested file
+        self.tmpdir = tempfile.TemporaryDirectory()
+
+        # directory and file
+        self.dir1_name = "testdir1"
+        self.dir2_name = "testdir2"
+        self.dir3_name = "testdir3"
+        self.testfile1_name = "testfile1"
+        self.testfile2_name = "testfile2"
+        self.testfile3_name = "testfile3"
+        self.testfile4_name = "testfile4"
+
+        # paths used in making outside.zip
+        dir1_path = os.path.join(self.tmpdir.name, self.dir1_name)
+        dir2_path = os.path.join(dir1_path, self.dir2_name)
+        dir3_path = os.path.join(self.tmpdir.name, self.dir3_name)
+        testfile1_path = os.path.join(dir1_path, self.testfile1_name)
+        testfile2_path = os.path.join(dir2_path, self.testfile2_name)
+        testfile3_path = os.path.join(dir3_path, self.testfile3_name)
+        testfile4_path = os.path.join(self.tmpdir.name, self.testfile4_name)
+
+        # paths used in tests
+        for dir in [dir1_path, dir2_path, dir3_path]:
+            os.mkdir(dir)
+
+        for file_path in [testfile1_path, testfile2_path,
+                          testfile3_path, testfile4_path]:
+            with open(file_path, "w") as f:
+                f.write(self.test_string)
+
+        # create zip without directory
+        self.pwd = os.getcwd()
+        os.chdir(self.tmpdir.name)
+        cmd = ["zip", "-rD", self.zip_file_name, "."]
+
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+
+        assert stderr == b""
+
+    def tearDown(self):
+        os.chdir(self.pwd)
+        self.tmpdir.cleanup()
+
+    def test_list(self):
+        with self.fs_handler.open_as_container(self.zip_file_name) as handler:
+            cases = [
```

Comment:
could you separate a test function in each case?

In the current version, it cannot be determined in which case failed. Moreover, if one case fails, subsequent cases will not be executed.